File: frontend/src/types/evaluation.ts
export interface EvaluationSummary {
  evaluation_name: string;
  dataset: { type: string; role: string; total_records: number; train_records: number; validation_records: number; test_records: number };
  baseline: { net_recovery: number; recovery_rate: number };
  unconstrained_ml: { net_recovery: number; improvement: number; relative_improvement_percent: number; constraint_status: string };
  recoveriq: { net_recovery: number; improvement: number; relative_improvement_percent: number; recovery_rate: number; solver_status: string };
  constraints: { total_recovery_spend: number; incentive_budget: number; retry_capacity: number; contact_capacity: number; whatsapp_capacity: number; max_incentive_actions: number; max_human_escalations: number; solver_timeout_ms: number };
  scale_benchmarks: { payments: number; feature_prep_seconds: number; ml_scoring_seconds: number; policy_economics_seconds: number; solver_seconds: number; total_seconds: number; status: string }[];
}

export interface EvaluationAssignment {
  payment_id: string; strategy: string; action: string; amount: number;
  true_recovery_probability: number; expected_gross_revenue: number; expected_net_revenue: number;
  intervention_cost: number; incentive_cost: number;
}

export interface EvaluationAssignmentPage {
  artifact: string; columns: string[]; items: EvaluationAssignment[]; page: number; page_size: number; total: number;
}

File: backend/app/schemas/evaluation.py
from pydantic import BaseModel, Field


class EvaluationDataset(BaseModel):
    type: str
    role: str
    total_records: int
    train_records: int
    validation_records: int
    test_records: int


class BaselineEvaluation(BaseModel):
    net_recovery: float
    recovery_rate: float


class UnconstrainedEvaluation(BaseModel):
    net_recovery: float
    improvement: float
    relative_improvement_percent: float
    constraint_status: str


class RecoverIQEvaluation(BaseModel):
    net_recovery: float
    improvement: float
    relative_improvement_percent: float
    recovery_rate: float
    solver_status: str


class EvaluationConstraints(BaseModel):
    total_recovery_spend: float
    incentive_budget: float
    retry_capacity: int
    contact_capacity: int
    whatsapp_capacity: int
    max_incentive_actions: int
    max_human_escalations: int
    solver_timeout_ms: int


class ScaleBenchmark(BaseModel):
    payments: int
    feature_prep_seconds: float
    ml_scoring_seconds: float
    policy_economics_seconds: float
    solver_seconds: float
    total_seconds: float
    status: str


class EvaluationSummary(BaseModel):
    evaluation_name: str
    dataset: EvaluationDataset
    baseline: BaselineEvaluation
    unconstrained_ml: UnconstrainedEvaluation
    recoveriq: RecoverIQEvaluation
    constraints: EvaluationConstraints
    scale_benchmarks: list[ScaleBenchmark]


class EvaluationAssignment(BaseModel):
    payment_id: str
    strategy: str
    action: str
    amount: float
    true_recovery_probability: float
    expected_gross_revenue: float
    expected_net_revenue: float
    intervention_cost: float
    incentive_cost: float


class EvaluationAssignmentPage(BaseModel):
    artifact: str
    columns: list[str]
    items: list[EvaluationAssignment]
    page: int = Field(ge=1)
    page_size: int = Field(ge=1)
    total: int

